### PR TITLE
Update Python interop tests to use google-auth package

### PR DESCRIPTION
**Pull request by kpayson64**
We can't fully remove oauth2client yet as it is used in a beta api unit test.

### CODE REVIEW

File: tools/run_tests/helper_scripts/build_python.sh [hunk 1]
```diff
@@ -187,7 +187,7 @@ $VENV_PYTHON $ROOT/src/python/grpcio_reflection/setup.py build_package_protos
 pip_install_dir $ROOT/src/python/grpcio_reflection
 
 # Build/install tests
-$VENV_PYTHON -m pip install coverage oauth2client
+$VENV_PYTHON -m pip install coverage oauth2client google-auth requests
```
Comment (nathanielmanistaatgoogle) [inline tools/run_tests/helper_scripts/build_python.sh]: We don't have these pinned? Huh. Okay. Is now the time to think about pinning them?

File: src/python/grpcio_tests/tests/interop/client.py [hunk 2]
```diff
@@ -29,10 +29,11 @@
 """The Python implementation of the GRPC interoperability test client."""
 
 import argparse
-from oauth2client import client as oauth2client_client
+import os
 
+import google.auth
```
Comment (nathanielmanistaatgoogle) [inline src/python/grpcio_tests/tests/interop/client.py]: [`from google import auth`](https://google.github.io/styleguide/pyguide.html?showone=Imports#Imports)
[`from google.auth import jwt`](https://google.github.io/styleguide/pyguide.html?showone=Imports#Imports)

(And throughout. You may want to toss in aliases like `grpc_auth_transport`, `jwt_auth`, `auth_environment_variables`, and so forth.)

File: tools/run_tests/helper_scripts/build_python.sh (hunk 1 above)
Comment (kpayson64) [inline tools/run_tests/helper_scripts/build_python.sh]: Done

File: src/python/grpcio_tests/tests/interop/client.py (hunk 2 above)
Comment (kpayson64) [inline src/python/grpcio_tests/tests/interop/client.py]: Done

**Comment by kpayson64** [comment]
@nathanielmanistaatgoogle Addressed comments, back to you

**Comment by kpayson64** [comment]
https://github.com/grpc/grpc/issues/11105
#11123 
#9088

**Pull request merged**